webapp/src/components/issue_attribute_selector.jsx

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import ReactSelect from 'react-select';
+import PropTypes from 'prop-types';
+
+import {getStyleForReactSelect} from 'utils/styles';
+import Setting from 'components/setting';
+
+export default class IssueAttributeSelector extends PureComponent {
+    static propTypes = {
+        isMulti: PropTypes.bool.isRequired,
+        repo: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        onChange: PropTypes.func.isRequired,
+        loadOptions: PropTypes.func.isRequired,
+        selection: PropTypes.oneOfType([
+            PropTypes.array,
+            PropTypes.object,
+        ]).isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            options: [],
+            isLoading: false,
+            error: null,
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.repo) {
+            this.loadOptions();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.repo && prevProps.repo !== this.props.repo) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions = async () => {
+        this.setState({isLoading: true});
+
+        try {
+            const options = await this.props.loadOptions();
+            this.filterSelection(options);
+            this.setState({
+                options,
+                isLoading: false,
+                error: null,
+            });
+        } catch (err) {
+            this.filterSelection([]);
+            this.setState({
+                error: err,
+                isLoading: false,
+            });
+        }
+    };
+
+    filterSelection = (options) => {
+        if (this.props.isMulti) {
+            const filtered = options.filter((option) => this.props.selection.includes(option.value));
+            this.props.onChange(filtered);
+            return;
+        }
+
+        if (!this.props.selection) {
+            this.props.onChange({});
```

Comment:
Does this make sense to pass an object here? What assumptions is the parent making based on this?

Reply:
@mickmister Good point, I think it'd make more sense to pass `null`, updated: https://github.com/mattermost/mattermost-plugin-github/pull/343/commits/019f192ea44d38ae3f8902616489b1cfe4ad35ad